### PR TITLE
DEVPROD-9951 Agents request more often when they approach max idle time

### PR DESCRIPTION
**Pull request by ZackarySantana**
DEVPROD-9951

### Description
This adds the host idle time to the next task response and uses that to schedule the next task request sooner if it is approaching the time it will be marked/terminated as idle.

### CODE REVIEW

File: agent/agent.go [hunk 1]
```diff
@@ -282,18 +283,30 @@ func (a *Agent) loop(ctx context.Context) error {
...
+				// Sometimes the MaxIdleDuration is 0 if the app server
+				// fails to find the distro's idle timeout and the
+				// scheduler's global idle timeout.
+				if nextTask.MaxIdleDuration != 0 {
+					// DurationLeft is the time left until the host will be considered idle.
+					durationLeft := nextTask.MaxIdleDuration - time.Since(start)
+					// If the amount of time left is less than the next sleep internval, cut down on the sleep interval.
+					if durationLeft < sleepTime*2 {
+						// This guarantees that the agent will try to get a new task before the host is considered idle.
+						sleepTime = durationLeft / 2
```
[the remark this answers is not in the document]
Comment (ZackarySantana) [reply agent/agent.go:296]: Good idea!

File: agent/agent.go [hunk 2]
```diff
@@ -282,18 +283,30 @@ func (a *Agent) loop(ctx context.Context) error {
...
+					}
+				}
+				grip.Debugf("Agent found no task to run, sleeping %s.", sleepTime)
+				timer.Reset(sleepTime)
+				agentSleepInterval = agentSleepInterval * 2 // 10s, 20s, 40s/ 80s
```
[the remark this answers is not in the document]
Comment (ZackarySantana) [reply agent/agent.go:301]: This is a leftover comment from me thinking about it, I think we can just remove it too!

**Comment by ZackarySantana** [comment]
Sorry for the later review!

**Review by Kimchelly — APPROVED** [review]
LGTM mod one nit.

**Pull request merged**